src/modules/apps/e2e/report.tsx

```diff
@@ -0,0 +1,270 @@
+        {report.error && <FailedSpecDetail label={'Error'} text={report.error} indented={true} />}
+        {...pathOr<SpecTestReport[], SpecTestReport[]>([] as SpecTestReport[], ['report', 'tests'], report)
+          .filter(({ state }) => state !== 'passed')
+          .map(({ testId, title, body, stack, error }) => {
+            const _screenshots = screenshots.filter(ss => ss.testId === testId)
+            return (
+              <Box key={title.join('')} flexDirection="column">
+                <FailedSpecDetail label={'Test'} text={title.join(' ')} indented={false} />
+                {body && <FailedSpecDetail label={'Body'} text={body} indented={true} />}
+                {stack && <FailedSpecDetail label={'Stack'} text={stack} indented={true} />}
+                {error && <FailedSpecDetail label={'Error'} text={error} indented={true} />}
+                {_screenshots.length > 0 && (
+                  <FailedSpecDetail
+                    label={'Screenshots'}
+                    text={_screenshots.map(ss => ' ' + ss.path).join('\n')}
+                    indented={_screenshots.length > 1}
+                  />
+                )}
+              </Box>
+            )
+          })}
+        {video && <FailedSpecDetail label={'Video'} text={video} indented={false} />}
+      </Box>
+    </Box>
+  )
+}
+
+const completedAppColors = (fail: [string, SpecReport][]) => {
+  if (fail.length === 0) return { bgGreen: true, black: true }
+  return { bgRed: true, white: true }
+}
+
+type AppProps = {
+  appId: string
+  specs: AppReport
+}
+
+const Completed: React.FunctionComponent<AppProps> = ({ appId, specs }) => {
+  const specsReports = toPairs(specs)
+  const { fail = [] } = groupBy(([, sr]) => (sr.state === 'passed' ? 'pass' : 'fail'), specsReports)
+  const failed = fail.length > 0
+  return (
+    <Box flexDirection="column">
+      <Box>
+        <Color {...completedAppColors(fail)}>{failed ? ' FAIL ' : ' PASS '}</Color>
+        <Box marginLeft={1}>
+          <AppId appId={appId} />
+        </Box>
+      </Box>
+      <Box flexDirection="column" marginLeft={2}>
+        {fail.map(([spec, sr]) => (
+          <FailedSpec key={spec} spec={spec} report={sr} />
+        ))}
+      </Box>
+    </Box>
+  )
+}
+
+const Running: React.FunctionComponent<AppProps> = ({ appId, specs }) => {
+  const specsReports = values(specs)
+  const { running = specsReports.length, completed = 0 } = countBy(
+    sr => (completedSpec(sr) ? 'completed' : 'running'),
+    specsReports
+  )
+  return (
+    <Box>
+      <Color bgYellow black>
+        {' RUNS '}
+      </Color>
+      <Box marginLeft={1}>
+        <AppId appId={appId} />
+      </Box>
+      <Box marginLeft={1}>
+        <Color greenBright>{`${completed} specs completed`}</Color>
+        <Text>{`, ${running + completed} total`}</Text>
+      </Box>
+    </Box>
+  )
+}
+
+type ReportProps = {
+  completed: Array<[string, AppReport]>
+  running: Array<[string, AppReport]>
```

Comment:
Using array to send data like this is not that cool - we don't have vscode autocomplete this way, and we don't know just by seeing this type what is this